From 2f37eaa39c6fb845c181281d5708b3413cf29b79 Mon Sep 17 00:00:00 2001
From: Aditya Kumar Singh <quic_adisi@quicinc.com>
Date: Wed, 6 Sep 2023 11:10:35 +0530
Subject: [PATCH] iw: print NO-EHT flags for reg get command

If a reg rule is marked with NO-EHT flag, print "NO-EHT" while
displaying the reg rules via reg get command.

Signed-off-by: Aditya Kumar Singh <quic_adisi@quicinc.com>
---
 nl80211.h | 2 ++
 reg.c     | 1 +
 2 files changed, 3 insertions(+)

diff --git a/nl80211.h b/nl80211.h
index f3f456094989..db431b13776d 100644
--- a/nl80211.h
+++ b/nl80211.h
@@ -4344,6 +4344,7 @@ enum nl80211_sched_scan_match_attr {
  * @NL80211_RRF_NO_HE: HE operation not allowed
  * @NL80211_RRF_NO_320MHZ: 320MHz operation not allowed
  * @NL80211_RRF_PSD: channels has power spectral density value
+ * @NL80211_RRF_NO_EHT: EHT operation not allowed
  */
 enum nl80211_reg_rule_flags {
 	NL80211_RRF_NO_OFDM		= 1<<0,
@@ -4364,6 +4365,7 @@ enum nl80211_reg_rule_flags {
 	NL80211_RRF_NO_HE		= 1<<17,
 	NL80211_RRF_NO_320MHZ		= 1<<18,
 	NL80211_RRF_PSD                 = 1<<19,
+	NL80211_RRF_NO_EHT              = 1<<20,
 };
 
 #define NL80211_RRF_PASSIVE_SCAN	NL80211_RRF_NO_IR
diff --git a/reg.c b/reg.c
index 1ff4fe998a18..2c3bffb95486 100644
--- a/reg.c
+++ b/reg.c
@@ -251,6 +251,7 @@ static int print_reg_handler(struct nl_msg *msg, void *arg)
 		PARSE_FLAG(NL80211_RRF_NO_160MHZ, "NO-160MHZ");
 		PARSE_FLAG(NL80211_RRF_NO_HE, "NO-HE");
 		PARSE_FLAG(NL80211_RRF_NO_320MHZ, "NO-320MHZ");
+		PARSE_FLAG(NL80211_RRF_NO_EHT, "NO-EHT");
 
 		/* Kernels that support NO_IR always turn on both flags */
 		if ((flags & NL80211_RRF_NO_IR) && (flags & __NL80211_RRF_NO_IBSS)) {
-- 
2.17.1

